File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);
        int ans=0;
        for( int cost:costs){
            if (cost<=coins){
                ans+=1;
                coins-=cost;
            }
        }
        return ans;
    }
}

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        n=0
        p=0
        for c in costs:
            if c<=coins:
                p+=c
                if p>coins:
                    break

                n+=1     
        return n

